src/test/java/org/opentripplanner/middleware/BugsnagReportingTest.java

```diff
@@ -0,0 +1,25 @@
+package org.opentripplanner.middleware;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.bugsnag.BugsnagReporter;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for Bugsnag reporting. All are set to @Disabled because a BUGSNAG_PROJECT_NOTIFIER_API_KEY is required in the
+ * test\env.yml file which gets committed. If these tests are required to be run, the README explains how to get a
+ * BUGSNAG_PROJECT_NOTIFIER_API_KEY from Bugsnag which can be temporarily saved to test\env.yml.
+ */
+public class BugsnagReportingTest extends OtpMiddlewareTest {
+
+    @Test @Disabled
+    public void createErrorReportWithExceptionAndMessage() {
+        assertTrue(BugsnagReporter.reportErrorToBugsnag(new Exception("Unit test exception 1"), "Unit test 1"));
+    }
+
+    @Test @Disabled
+    public void createErrorReportWithException() {
+        assertTrue(BugsnagReporter.reportErrorToBugsnag(new Exception("Unit test exception 2"), null));
+    }
```

Comment:
Is it possible to create a test that triggers a report somewhere in the actual code (Persistence perhaps)?

Reply:
Would it make sense to create an `/makeerror` endpoint that generates a 500 error so we can catch it?

Reply:
@landonreed if @binh-dam-ibigroup suggestion it ok, I will implement.

Reply:
Hmm, perhaps. I don't think we would really want this to be an open HTTP endpoint though (only accessible by Admin users). I'm kind of ambivalent here because this doesn't seem like something we really want to build into the non-test software, but it would be convenient to have a way to verify that things are configured correctly. @evansiroky?

Reply:
It will probably fail if you just try to insert a `null` TripRequest, right? That seems like an easy test to write up.

Reply:
@landonreed - I've made an update and created a unit test. As you will see this required additional code within the TypedPersistence -> create method and a less conventional unit test. Now that I have done this other methods in the TypedPersistence class could be updated in a similar way, but would like your view on this.